azure-core/azure/core/pipeline/policies/base.py

```diff
@@ -71,17 +72,16 @@ def on_request(self, request, **kwargs):
         # type: (PipelineRequest[HTTPRequestType], Any) -> None
         """Is executed before sending the request to next policy.
         """
-        pass
 
     def on_response(self, request, response, **kwargs):
         # type: (PipelineRequest[HTTPRequestType], PipelineResponse[HTTPRequestType, HTTPResponseType], Any) -> None
         """Is executed after the request comes back from the policy.
         """
-        pass
 
-    def on_exception(self, request, **kwargs):
+    #pylint: disable=no-self-use
```

Comment:
This is certainly a matter for others to weigh in, but personally I prefer to mark unused args with a leading underscore, i.e.
```
 def on_exception(_self, _request, **kwargs):
```
rather than with pragmas, but there may be guidelines specific to this codebase I'm unaware of

Reply:
i like this idea, although self would still remain self (or else pylint throws a different error)